modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/EditFileEntryMVCActionCommand.java

```diff
@@ -310,15 +310,14 @@ else if (cmd.equals(Constants.PREVIEW)) {
 					sendRedirect(actionRequest, actionResponse, redirect);
 				}
 				else {
-					redirect = _portal.escapeRedirect(
-						ParamUtil.getString(actionRequest, "redirect"));
+					redirect = ParamUtil.getString(actionRequest, "redirect");
 
 					if (Validator.isNotNull(redirect)) {
-						if (cmd.equals(Constants.ADD) && (fileEntry != null)) {
-							String portletResource =
-								HttpComponentsUtil.getParameter(
-									redirect, "portletResource", false);
+						String portletResource =
+							HttpComponentsUtil.getParameter(
+								redirect, "portletResource", false);
 
+						if (cmd.equals(Constants.ADD) && (fileEntry != null)) {
```

Comment:
I would add the condition `Validator.isNotNull(portletResource)` to this IF statement, to remove the inner IF statement.

Reply:
I'd prefer to keep it as it was in order to avoid noise in the PR.

The only reason why I touch this was to extract the portletResource variable and use it outside this if clause

Reply:
Not a fan of leaving it like this, but OK, let's move on.

Reply:
The thing is that, as part of a bug fix I tend to think that the lesser the better.

If we introduce this kind of refactoring the focus of the main fix could be lost (and we always have the risk of introduce regressions - If we had the infrastructure to do functional tests I could be more confident of this kind of changes, but for now I'd prefer to be as conservative as possible).